homeassistant/components/vallox/sensor.py

```diff
@@ -117,53 +126,29 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 class ValloxSensor(SensorEntity):
     """Representation of a Vallox sensor."""
 
+    _attr_should_poll = False
+
     def __init__(
-        self, name, state_proxy, metric_key, device_class, unit_of_measurement, icon
+        self,
+        name,
+        state_proxy,
+        metric_key,
+        device_class,
+        state_class,
+        unit_of_measurement,
+        icon,
```

Comment:
You might want to look into using `SensorEntityDescription` classes, which can really reduce this code a bit more.